e2-studio-star-rx72n-firmware/libs/rx_hcsr04/src/rx_hcsr04.c

```
@@ -1564,22 +1881,35 @@ rx_err_t rx_hcsr04_init(rx_hcsr04_t* handle, const rx_hcsr04_config_t* config)
     return err;
   }
 
-  /* Configure echo pin as input */
-  err = hcsr04_hal_gpio_set_input(config->echo_pin);
-  if (err != k_rx_ok) {
-    /* Cleanup trigger pin */
-    (void)hcsr04_hal_gpio_deinit(config->trigger_pin); /* Cleanup, ignore errors */
-    return err;
+  /* Configure echo pin based on mode */
+  if (config->echo_mode == k_hcsr04_echo_irq) {
```

Clang (14.0.6)

[warning] if with identical then and else branches

(bugprone-branch-clone)